vars/customWARPackager.groovy

```diff
@@ -0,0 +1,117 @@
+    def version = null
+    def artifactId = null
+    if (bomFilePath == null) {
+        def files = findFiles(glob: "bom.yml")
+        if (files != null && files.length > 0) {
+            echo "BOM file is not explicitly defined, but there is bom.yml in the root. Using it"
+            bomFilePath = files[0]
+        }
+    }
+    if (bomFilePath != null ) {
+        def bom = readYaml(file: bomFilePath)
+        //TODO: adjust if parameters are made optional
+        if (bom.metadata.labels != null) {
+            version = bom.metadata.labels.version
+            artifactId = bom.metadata.labels.artifactId
+        }
+    }
+
+    if ((version == null || artifactId == null) && configFilePath != null) { // Try config YML
+        def config = readYaml(file: configFilePath)
+        if (config.bundle != null && artifactId == null) {
+            artifactId = config.bundle.artifactId
+        }
+        if (config.buildSettings != null && version == null) {
+            version = config.buildSettings.version
+        }
+    }
+
+    if (version == null) {
+        version = "1.0-SNAPSHOT"
+    }
+    if (artifactId == null) {
+        artifactId = "jenkins-war"
+    }
+
+    // Warm-up
+    // TODO: Remove once JENKINS-51070 is fixed
+    List<String> mavenWarmup = [
+        '--batch-mode', '--errors',
+        "org.apache.maven.plugins:maven-dependency-plugin:3.0.2:get",
+        "-Dartifact=io.jenkins.tools.custom-war-packager:custom-war-packager-cli:${cwpVersion}",
+        "-Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn"
+    ]
+    infra.runMaven(mavenWarmup, jdk, null, mvnSettingsFile)
+
+    echo "Downloading Custom WAR Packager CLI ${cwpVersion}"
+    List<String> mavenOptions = [
+        '--batch-mode', '--errors',
+        "com.googlecode.maven-download-plugin:download-maven-plugin:1.4.0:artifact",
+        "-DgroupId=io.jenkins.tools.custom-war-packager",
+        "-DartifactId=custom-war-packager-cli",
+        "-Dclassifier=jar-with-dependencies",
+        "-Dversion=${cwpVersion}",
+        "-DoutputDirectory=${pwd()}",
+        "-DoutputFileName=cwp-cli.jar",
+        "-Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn"
+    ]
+    infra.runMaven(mavenOptions, jdk, null, mvnSettingsFile)
+
+    def command = "java -jar cwp-cli.jar --batch-mode -configPath ${configFilePath} -version=${version}"
+    if (environment != null) {
+        command += " --environment ${environment}"
+    }
+    if (bomFilePath != null) {
+        command += " --bomPath ${bomFilePath}"
+    }
+    if (mvnSettingsFile != null) {
+        command += " -mvnSettingsFile=${mvnSettingsFile}"
+    }
+    infra.runWithMaven(command)
+
+    dir("tmp/output/target/") {
+        archiveArtifacts artifacts: "${artifactId}-${version}.war"
+        archiveArtifacts artifacts: "${artifactId}-${version}.bom.yml"
```

Comment:
collapse into one step

Reply:
It's not regexp unfortunately. There are other artifacts, and I am not sure how to filter it in Glob easily